### PR TITLE
fix: base64url encode/decode

### PR DESCRIPTION
**Pull request by ragnika**

### CODE REVIEW

**Comment by vijayshiyani** [comment]
@ragnika 
Is this implementation identical to the one in the 'jose' library? It appears fine.

is it possible to avoid specific runtime implementations and instead detect whether it's a browser or Node environment within the functions?
e.g. 

`
let base64;
  if (typeof btoa === 'function') {
    // Browser environment
    base64 = btoa(input);
  } else {
    // Node.js environment
    base64 = Buffer.from(input).toString('base64');
  }
  
  `

**Review by vijayshiyani — APPROVED** [review]

**Comment by ragnika** [comment]
> @ragnika Is this implementation identical to the one in the 'jose' library? It appears fine.
> 
> is it possible to avoid specific runtime implementations and instead detect whether it's a browser or Node environment within the functions? e.g.
> 
> ` let base64; if (typeof btoa === 'function') { // Browser environment base64 = btoa(input); } else { // Node.js environment base64 = Buffer.from(input).toString('base64'); }
> 
> `

https://nodejs.org/api/buffer.html#buffer_static_method_buffer_from_array

btoa and atob are technically in the global namespace in node but are not recommended for use

**Comment by linasi** [comment]
> > @ragnika Is this implementation identical to the one in the 'jose' library? It appears fine.
> > is it possible to avoid specific runtime implementations and instead detect whether it's a browser or Node environment within the functions? e.g.
> > ` let base64; if (typeof btoa === 'function') { // Browser environment base64 = btoa(input); } else { // Node.js environment base64 = Buffer.from(input).toString('base64'); }
> > `
> 
> https://nodejs.org/api/buffer.html#buffer_static_method_buffer_from_array
> 
> btoa and atob are technically in the global namespace in node but are not recommended for use

Also, `btoa` works with ASCII only while we need to support `UTF-8`.

File: src/runtime/base64url.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { Buffer } from 'buffer';
```
Comment (linasi) [inline src/runtime/base64url.ts:1]: @ragnika what is this file's purpose? there is one under `node` directory already.

File: src/runtime/browser/base64url.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
+const encoder = new TextEncoder();
+const decoder = new TextDecoder();
+
+export const encodeBase64 = (input: Uint8Array | string) => {
```
Comment (linasi) [inline src/runtime/browser/base64url.ts:4]: @ragnika I see this implementation is borrowed from `jose` library. Does it handle utf-8?

**Review by linasi — APPROVED** [review]
👍

File: src/runtime/browser/base64url.ts (hunk 2 above)
Comment (ragnika) [reply src/runtime/browser/base64url.ts:4]: @linasi yes. TextEncoder emits a stream of utf-8 bytes

https://developer.mozilla.org/en-US/docs/Web/API/TextEncoder

File: src/runtime/base64url.ts (hunk 1 above)
Comment (ragnika) [reply src/runtime/base64url.ts:1]: oops, this one is copied locally by running npm run dev:setup.
let me remove this

**Pull request merged**